starknet_py/net/full_node_client.py

```diff
@@ -361,6 +362,58 @@ async def estimate_fee(
             ),
         )
 
+    async def estimate_message_fee(
+        self,
+        from_address: str,
+        to_address: Hash,
+        entry_point_selector: Hash,
+        payload: List[Hash],
+        block_hash: Optional[Union[Hash, Tag]] = None,
+        block_number: Optional[Union[int, Tag]] = None,
+    ) -> EstimatedFee:
+        # pylint: disable=too-many-arguments
+        """
+        :param from_address: The address of the L1 (Ethereum) contract sending the message.
+        :param to_address: The target L2 (Starknet) address the message is sent to.
+        :param entry_point_selector: The selector of the l1_handler in invoke in the target contract.
+        :param payload: Payload of the message.
+        :param block_hash: Hash of the requested block or literals `"pending"` or `"latest"`.
+            Mutually exclusive with ``block_number`` parameter. If not provided, queries block `"pending"`.
+        :param block_number: Number (height) of the requested block or literals `"pending"` or `"latest"`.
+            Mutually exclusive with ``block_hash`` parameter. If not provided, queries block `"pending"`.
+        """
+        block_identifier = get_block_identifier(
+            block_hash=block_hash, block_number=block_number
+        )
+
+        assert _is_valid_eth_address(
+            from_address
+        ), f"Argument 'from_address': {from_address} is not a valid Ethereum address."
+
+        message_body = {
+            "from_address": from_address,
```

Comment:
It's a felt in spec, so I guess it should be converted too

Reply:
It is `ETH_ADDRESS`
```
"from_address": {
    "description": "The address of the L1 contract sending the message",
    "$ref": "#/components/schemas/ETH_ADDRESS"
},
```
which is
```
"ETH_ADDRESS": {
    "title": "Ethereum address",
    "type": "string",
    "$comment": "An ethereum address",
    "description": "an ethereum address represented as 40 hex digits",
    "pattern": "^0x[a-fA-F0-9]{40}$"
},
```